__builtin_available() is not supported by GCC.

Wrap in MAC_OS_X_VERSION_MAX_ALLOWED pre-processor conditional, assuming
builds on macOS 11+ will only use clang.

--- ./src/interface/toolbar.mm.orig	2023-03-25 20:47:43.000000000 +0100
+++ ./src/interface/toolbar.mm	2023-03-25 20:49:51.000000000 +0100
@@ -1,11 +1,14 @@
 #include <wx/wx.h>
 
+#include <AvailabilityMacros.h>
 #include <AppKit/NSWindow.h>
 
 void fix_toolbar_style(wxFrame& frame)
 {
+#if MAC_OS_X_VERSION_MAX_ALLOWED >= 110000
 	if (__builtin_available(macos 11.0, *)) {
 		WXWindow tlw = frame.MacGetTopLevelWindowRef();
 		[tlw setToolbarStyle:NSWindowToolbarStyleExpanded];
 	}
+#endif
 }
